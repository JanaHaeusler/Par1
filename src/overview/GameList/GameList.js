import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import Game from '../Game'

GameList.propTypes = {
    savedGameProfiles: PropTypes.object.isRequired,
    onDelete: PropTypes.func.isRequired,
    onEdit: PropTypes.func.isRequired,
    showCreatePage: PropTypes.func.isRequired,
}

export default function GameList({savedGameProfiles, onDelete, onEdit, showCreatePage}) {

  return(
    <GameListWrapper>
        {savedGameProfiles.allIds.map((id) => {
          const {location, date, players, winner, shots, _id} = savedGameProfiles.byId[id]
          const playersString = players.join(', ')
          return <Game 
                    key={_id} 
                    id={_id} 
                    onDelete={onDelete} 
                    onEdit={onEdit} 
                    location={location} 
                    date={date} 
                    players={playersString} 
                    winner={winner} 
                    shots={shots}
                    showCreatePage={showCreatePage}/>
        })}
    </GameListWrapper>
  )
}

const GameListWrapper = styled.div`
  display: grid;
  gap: 20px;
`