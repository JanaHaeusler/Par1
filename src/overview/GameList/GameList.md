```jsx
<GameList
  savedGameProfiles={{
    byId: {
      1: {
        location: 'Horner Racecourse',
        date: '2020-12-24',
        players: 'John, Jane',
        winner: 'Jane',
        shots: '89',
        _id: '1',
      },
      2: {
        location: 'City Park',
        date: '2020-01-01',
        players: 'Fritz, Hanni, Nanni',
        winner: 'Fritz',
        shots: '35',
        _id: '2',
      },
    },
    allIds: [1, 2],
  }}
  onDelete={() => {}}
  onEdit={() => {}}
  showCreatePage={() => {}}
/>
```
