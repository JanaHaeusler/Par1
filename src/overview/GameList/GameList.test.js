import GameList from './GameList'
import { render } from '@testing-library/react'

const testProps = {
  savedGameProfiles: {
    byId: {
      1: {
        location: 'Horner Racecourse',
        date: '2020-12-24',
        players: 'John, Jane',
        winner: 'Jane',
        shots: '89',
        _id: '1',
      },
      2: {
        location: 'City Park',
        date: '2020-01-01',
        players: 'Fritz, Hanni, Nanni',
        winner: 'Fritz',
        shots: '35',
        _id: '2',
      },
    },
    allIds: [1, 2],
  }, 
  onDelete: jest.fn(),
  onEdit: jest.fn(),
  showCreatePage: jest.fn(),
}

describe('GameList', () => {
  
    it('renders correctly', () => {
        const { container, getByText, getAllByTestId } = render(<GameList {...testProps}/>)
        
        expect(container.firstChild).toMatchSnapshot()

        expect(getByText('Horner Racecourse')).toBeInTheDocument()
        expect(getByText('2020-12-24')).toBeInTheDocument()
        expect(getByText('John, Jane')).toBeInTheDocument()
        expect(getByText('Jane')).toBeInTheDocument()
        expect(getByText('89')).toBeInTheDocument()
        
        expect(getByText('City Park')).toBeInTheDocument()
        expect(getByText('2020-01-01')).toBeInTheDocument()
        expect(getByText('Fritz, Hanni, Nanni')).toBeInTheDocument()
        expect(getByText('Fritz')).toBeInTheDocument()
        expect(getByText('35')).toBeInTheDocument()

        const ButtonDelete = getAllByTestId('button-set-delete')
        expect(ButtonDelete).toHaveLength(2)
        const ButtonEdit = getAllByTestId('button-edit')
        expect(ButtonEdit).toHaveLength(2)
      })
})