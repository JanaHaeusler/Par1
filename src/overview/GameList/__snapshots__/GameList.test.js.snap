// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameList renders correctly 1`] = `
.c1 {
  margin: 0 20px;
  padding: 10px;
  box-shadow: 0 0 10px var(--primary-medium);
  border-radius: 10px;
  background-color: var(--text-light);
  font-size: 0.9rem;
}

.c2 {
  display: grid;
  grid-template-rows: repeat(4,auto);
  grid-template-columns: 1fr 1fr;
  grid-gap: 10px;
}

.c4 {
  grid-column-start: 2;
  grid-row-start: 1;
  justify-self: end;
}

.c3 {
  grid-column: 1 / 3;
  grid-row-start: 2;
  text-transform: uppercase;
}

.c5 {
  grid-column: 1 / 4;
  grid-row-start: 3;
}

.c6 {
  grid-column-start: 1;
  grid-row-start: 4;
}

.c7 {
  grid-column-start: 2;
  grid-row-start: 4;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  width: 100%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 5px;
  padding: 0;
  width: 30px;
  border: none;
  background: none;
}

.c10 {
  margin-right: 3px;
  fill: var(--secondary-dark);
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 20px 5px 5px 5px;
  padding: 0;
  width: 30px;
  border: none;
  background: none;
}

.c12 {
  margin-right: 3px;
  fill: var(--secondary-dark);
}

.c0 {
  display: grid;
  gap: 20px;
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        Horner Racecourse
      </div>
      <div
        class="c4"
      >
        2020-12-24
      </div>
      <div
        class="c5"
      >
        <h4>
          Player(s)
        </h4>
        <span>
          John, Jane
        </span>
      </div>
      <div
        class="c6"
      >
        <h4>
          Winner(s)
        </h4>
        <span>
          Jane
        </span>
      </div>
      <div
        class="c7"
      >
        <h4>
          Total Shots
        </h4>
        <span>
          89
        </span>
      </div>
    </div>
    <div
      class="c8"
    >
      <button
        class="c9"
        data-testid="button-set-delete"
      >
        <svg
          class="c10"
        >
          bin.svg
        </svg>
      </button>
      <button
        class="c11"
        data-testid="button-edit"
      >
        <svg
          class="c12"
        >
          pencil.svg
        </svg>
      </button>
    </div>
  </section>
  <section
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        City Park
      </div>
      <div
        class="c4"
      >
        2020-01-01
      </div>
      <div
        class="c5"
      >
        <h4>
          Player(s)
        </h4>
        <span>
          Fritz, Hanni, Nanni
        </span>
      </div>
      <div
        class="c6"
      >
        <h4>
          Winner(s)
        </h4>
        <span>
          Fritz
        </span>
      </div>
      <div
        class="c7"
      >
        <h4>
          Total Shots
        </h4>
        <span>
          35
        </span>
      </div>
    </div>
    <div
      class="c8"
    >
      <button
        class="c9"
        data-testid="button-set-delete"
      >
        <svg
          class="c10"
        >
          bin.svg
        </svg>
      </button>
      <button
        class="c11"
        data-testid="button-edit"
      >
        <svg
          class="c12"
        >
          pencil.svg
        </svg>
      </button>
    </div>
  </section>
</div>
`;
